### PR TITLE
Added data source for dynamic secrets

### PR DESCRIPTION
**Pull request by maxcoulombe**
### :hammer_and_wrench: Description

Added the ability to "open" dynamic secrets, that is create a new credential instance on every run.

Demo (all sensitive values have been revoked):

https://github.com/hashicorp/terraform-provider-hcp/assets/109547106/b5c45609-2758-45ff-a610-fabbed54414e

Open question:
* Can we or should we not save anything in the statefile if they are regenerated on every run? The behavior is correct, it's just that the statefile diff is noisy and doesn't provide value

### :building_construction: Acceptance tests

- [x] Are there any feature flags that are required to use this functionality?
- [x] Have you added an acceptance test for the functionality being added?
- [x] Have you run the acceptance tests on this branch?

Output from acceptance testing:

```sh
$ 15:30:09 max:~/projects-cloud/terraform-provider-hcp$ make testacc TESTARGS='-run=TestAcc_dataSourceVaultSecretsDynamicSecret'
(...)
=== RUN   TestAcc_dataSourceVaultSecretsDynamicSecret
--- PASS: TestAcc_dataSourceVaultSecretsDynamicSecret (5.61s)
PASS
ok  	github.com/hashicorp/terraform-provider-hcp/internal/provider/vaultsecrets	5.622s
(...)
...

### CODE REVIEW

File: docs/data-sources/vault_secrets_dynamic_secret.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# hcp_vault_secrets_dynamic_secret (Data Source)
+
+-> **Note:** This data source is currently in beta.
+
+This data source generates a new dynamic secret instance.
+
+## Example Usage
+
+```terraform
+data "hcp_vault_secrets_rotating_secret" "example" {
```
Comment (austingebauer) [inline docs/data-sources/vault_secrets_dynamic_secret.md:17]: rotating -> dynamic?

File: templates/data-sources/vault_secrets_dynamic_secret.md.tmpl [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+# {{.Name}} ({{.Type}})
+
+-> **Note:** This data source is currently in beta.
+
+{{ .Description | trimspace }}
+
+## Example Usage
+
+{{ tffile "examples/data-sources/hcp_vault_secrets_rotating_secret/data-source.tf" }}
```
Comment (austingebauer) [inline templates/data-sources/vault_secrets_dynamic_secret.md.tmpl:16]: `hcp_vault_secrets_rotating_secret` -> `hcp_vault_secrets_dynamic_secret`

**Review by austingebauer — APPROVED** [review]
LGTM otherwise!

File: docs/data-sources/vault_secrets_dynamic_secret.md (hunk 1 above)
Comment (maxcoulombe) [reply docs/data-sources/vault_secrets_dynamic_secret.md:17]: Whoops, nice catch

**Review by jasonpilz — APPROVED** [review]

**Pull request merged**